src/rgw/driver/sfs/sqlite/sqlite_objects.cc

```diff
@@ -13,57 +13,57 @@
  */
 #include "sqlite_objects.h"
 
+#include "dbapi.h"
+#include "sqlite_query_utils.h"
 #include "sqlite_versioned_objects.h"
 
-using namespace sqlite_orm;
-
 namespace rgw::sal::sfs::sqlite {
 
 SQLiteObjects::SQLiteObjects(DBConnRef _conn) : conn(_conn) {}
 
 std::vector<DBObject> SQLiteObjects::get_objects(const std::string& bucket_id
 ) const {
-  auto storage = conn->get_storage();
-  return storage->get_all<DBObject>(
-      where(is_equal(&DBObject::bucket_id, bucket_id))
+  return GetSQLiteObjectsWhere<DBObject>(
+      conn->get(), "objects", "bucket_id", bucket_id
   );
 }
 
 std::optional<DBObject> SQLiteObjects::get_object(const uuid_d& uuid) const {
-  auto storage = conn->get_storage();
-  auto object = storage->get_pointer<DBObject>(uuid.to_string());
-  std::optional<DBObject> ret_value;
-  if (object) {
-    ret_value = *object;
-  }
-  return ret_value;
+  return GetSQLiteSingleObject<DBObject>(
+      conn->get(), "objects", "uuid", uuid.to_string()
+  );
 }
 
 std::optional<DBObject> SQLiteObjects::get_object(
     const std::string& bucket_id, const std::string& object_name
 ) const {
-  auto storage = conn->get_storage();
-  auto objects = storage->get_all<DBObject>(where(
-      is_equal(&DBObject::bucket_id, bucket_id) and
-      is_equal(&DBObject::name, object_name)
-  ));
-
-  std::optional<DBObject> ret_value;
-  // value must be unique
-  if (objects.size() == 1) {
-    ret_value = objects[0];
+  auto rows =
+      conn->get()
+      << R"sql(SELECT * FROM objects WHERE bucket_id = ? AND name = ?;)sql"
+      << bucket_id << object_name;
+  std::optional<DBObject> ret_object;
+  for (auto&& row : rows) {
+    ret_object = DBObject(row);
```

Comment:
`std::move(row)` otherwise you are calling the non-move constructor of DBObject (hopefully move constructor of DBObject could be more fast than the classical copy constructor)

Reply:
`std::move` just does a cast. We also would need to add the constructor that has the rvalue, otherwise it would still call the copy one. 

Good point, though, I will experiment and see if we can avoid copies.

The && is here as an universal reference which accepts everything.

Reply:
> std::move just does a cast.

These considerations because when you read `auto&& row` you'd expect the `row` reference to be then bound with move enabled functions/constructor/methods.
Otherwise no need to use the `&&` notation in the for.  

> Good point, though, I will experiment and see if we can avoid copies.

yup, this considerations are valid only if we support a `DBObject(.. &&row) `constructor (and if with that implementation we can obtain some advantages)

Reply:
Regarding the move semantics for all conversions... I've done a few experiments and checked the sqlite modern cpp library and I think we can get use of moves, but we'll still need extra work for taking the benefit in the conversion utils file. 
Right now we're not using move semantics for the optional assignment values, but I guess it could be beneficial in the future.
We said to explore this in a future PR, when the whole port to sqlite modern cpp is finished. 

Anyway... as the rows are defined with the && notation I'm using std::move and changed the constructors involved.